src/app/zap-templates/templates/app/CHIPClusters-src.zapt

```diff
@@ -38,12 +41,12 @@ CHIP_ERROR {{asCamelCased clusterName false}}Cluster::{{asCamelCased name false}
     // Command takes no arguments.
 {{/chip_server_cluster_command_arguments}}
 
-    ReturnErrorOnFailure(ZCLcommand->FinishCommand());
+    ReturnErrorOnFailure(mpCommandSender->FinishCommand());
 
     // #6308: This is a temporary solution before we fully support IM on application side and should be replaced by IMDelegate.
-    mDevice->AddIMResponseHandler(onSuccessCallback, onFailureCallback);
+    mDevice->AddIMResponseHandler(mpCommandSender, onSuccessCallback, onFailureCallback);
 
-    return mDevice->SendCommands();
+    return mDevice->SendCommands(mpCommandSender);
```

Comment:
So this isn't actually batching up multiple commands into a single InvokeInteraction right? This is still very much, 1 command PER invoke interaction?

Reply:
Yes, when talking about "sending multiple commands", we need to 
1. Make it possible to have multiple transactions between controller and device.
2. The transactions can carry more than one commands.

This PR is focus on the first one, and the second will be implemented in another PR.

Reply:
I'd suggest holding off on 2, since as part of the proposal/prototype I'm writing up, I'm combining CommandSender/CommandHandler/Command into a single 'InvokeInteraction' class that will handle:
* Permitting multiple commands to be bundled in a single invoke
* Allow for async responses to be sent
* Permit multiple invoke interactions over a single exchange AND InvokeInteraction object instance, to reduce allocation/de-allocation transients.

I'm also not certain what the real use-case for 2) is right now, since everything I've seen so far would only necessitate single-command-in-invoke interactions. Is there an example of it that is needed in the upcoming TEs?

Reply:
Block transfer for OTA comes to mind.